pktfwd/gps.go

```diff
@@ -14,19 +21,167 @@ import (
 	- if nothing available or set, get the coordinates from account server and use that in the status message
 */
 
-// enableGPS checks if there is an available GPS for this build - if yes,
-// tries to activate it.
-func enableGPS(ctx log.Interface, gpsPath string) (err error) {
-	if gpsPath == "" {
-		ctx.Warn("No GPS chip configured, ignoring")
-		return nil
+type GPS interface {
+	Start() error
+	Stop()
+	GetCoordinates() (*gateway.GPSMetadata, error)
+	PacketTime(uplink router.UplinkMessage) (time.Time, error)
+}
+
+type halGPS struct {
+	ctx     log.Interface
+	BgCtx   context.Context
+	Cancel  context.CancelFunc
+	GPSPath string
+}
```

Comment:
No need to expose fields in an unexposed struct. Better safe than sorry when exposing it: use small caps by default